### DISCUSSION TITLE
Cursor Pagination

### DISCUSSION THREAD

**Discussion by fedeisas** [laravel/nova-issues#4979] [Q&A] [answered]
It would be awesome to have a `Resource` property that switches pagination mode from `offset` to `cursor` to ease the load on database servers with large tables.

Related
- https://github.com/laravel/nova-issues/issues/2345
- https://nova.laravel.com/docs/4.0/resources/#customizing-pagination

**Comment by crynobone** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/172966/198410490-8addcbc3-6a94-45b2-89c8-0ce69eb3f176.png)

This would mean that you no longer can sort the table by none unique attribute/field since cursor pagination explicitly required unique column. Definitely a huge limitation for existing applications and breaking changes.

**Reply by fedeisas** [reply]
Yes, I suspect that it could force sort with the resource's primary key or some timestamp.

My use case is a large `events` table (50m+ records) and running constant `COUNT`s on it seems wasteful. It loads really slowly.

> Definitely a huge limitation for existing applications and breaking changes.

If the default pagination mode continues to be same maybe we can avoid it being a breaking change.

**Reply by crynobone** [reply]
Again, the cursor pagination limitation applied and would cause breaking change and Nova's field `sortable()` to not able to work any longer. 

We not making these changes at the moment.

**Reply by BobbyBorisov** [reply]
Maaf, @crynobone what if this change is made with the notice that by using it the `sortable()` won't be able to work as intented? I think this is a fair tradeoff
